### DISCUSSION TITLE
can we backup from teleport version 9 and restore to teleport version 14

### DISCUSSION THREAD

**Discussion by weeband** [gravitational/teleport#34115] [Q&A] [answered]
for some background, we're too far away from latest stable version and we want to catch up, we're on v9 right now.
I have read the upgrading docs and it seems like it's possible with migrating for every major version, so I upgrade from 9 to 10, then to 11, 12, 13 and 14
so I've thought if teleport have backup & restore features, I should be able to dump all data with `tctl get all --with-secrets > state.yaml` and restore to a new teleport v14 cluster (with same cluster name / public domain) with extra args `--bootstrap state.yaml`  just I can rollout fast and rollback quickly if anything bad happen

I've try that and it fail, `tsh db connect` wont work and I mistook that as a bug (https://github.com/gravitational/teleport/issues/34058), 
but is that method possible?

**Comment by webvictim** [comment, marked as answer]
Unfortunately restoring a v9 backup to a v14 Teleport cluster is exactly the same as just upgrading straight from v9 to v14 (which shouldn't be done)

It may be easier for you to just deploy a fresh v14 cluster and migrate your nodes over.

If not, the assuming you can get a 1-2 hour maintenance window where Teleport can be down, you should be able to shut down all proxies and agents, scale down your auth service to just 1 replica, then upgrade that from 9 -> 10 -> 11 -> 12 -> 13 -> 14, shut it down, upgrade all other auths, proxies and agents to v14, then bring the whole cluster back online at v14. The main concern with version upgrades is database and role format migrations which are all handled by the auth service anyway - the other components are largely stateless and can be upgraded straight from 9 -> 14.

**Reply by weeband** [reply]
thank you for the helpful answer, I will try as suggested

**Reply by weeband** [reply]
sorry, one more question, if  I upgrade teleport cluster from 9 -> 10 -> 11 -> 12 -> 13 -> 14, does the teleport agent can use the same certificate generated by v9 or I need to renew the certificate (`tctl auth sign --format=db ...`) from v14 and change the cert on all db?
